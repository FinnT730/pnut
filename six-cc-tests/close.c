int hash(char_ptr s) {
  int hash;
  int ix;
  hash = 0;
  ix = 0;
  while (s[ix] != 0) {
    hash = s[ix] + (hash << 6) + (hash << 16) - hash;
    hash = hash & 65535; /* Most shells have 32-bit integers, so we have to make sure we don't "overflow" */
    ix++;
  }
  return ix;
}

int main() {
  int MAX_SIZE;
  int_ptr f;
  char_ptr s;
  int len;
  int h;
  MAX_SIZE = 200;
  s = malloc(MAX_SIZE);
  f = open("six-cc-tests/close.c", 0);
  len = read(f, s, MAX_SIZE - 1);
  s[len] = 0;
  close(f);
  printf("Read content: %s\n", s);
  printf("File descriptor: %s\n", f);
  printf("Quote: \"\n", len);
  printf("Backslash: \\\n", len);
  printf("Read len: %d\n", len);
  printf("Read result: %.*s\n", len, s);
  printf("hash: %d\n", hash(s));
}
