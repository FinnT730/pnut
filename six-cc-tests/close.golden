Read content: int hash(char_ptr s) {{  int hash;;  int ix;;  hash = 0;;  ix = 0;;  while (s[ix] != 0) {{    hash = s[ix] + (hash << 6) + (hash << 16) - hash;;    hash = hash & 65535; /* Most shells have 32-bit int
File descriptor: six-cc-tests/close.c
Quote: "
Backslash: \
Read len: 199
Read result: int hash(char_ptr s) {{  int hash;;  int ix;;  hash = 0;;  ix = 0;;  while (s[ix] != 0) {{    hash = s[ix] + (hash << 6) + (hash << 16) - hash;;    hash = hash & 65535; /* Most shells have 32-bit int
hash: 199
